test/Microsoft.AspNet.Mvc.ModelBinding.Test/Validation/AssociatedValidatorProviderTest.cs

```diff
@@ -76,6 +76,73 @@ public void GetValidatorsForPropertyWithMixedAttributes()
             Assert.True(callbackAttributes.Any(a => a is RequiredAttribute));
         }
 
+        [Fact]
+        public void GetModelMetadataTypeAttributeValidators_DifferentAttributesOnProperty()
+        {
+            // Arrange
+            IEnumerable<Attribute> callbackAttributes = null;
+            var metadata = _metadataProvider.GetMetadataForProperty(null, typeof(PropertyViewModel), "LocalAttributes");
+            Mock<TestableAssociatedValidatorProvider> provider = new Mock<TestableAssociatedValidatorProvider> { CallBase = true };
+            provider.Setup(p => p.AbstractGetValidators(metadata, It.IsAny<IEnumerable<Attribute>>()))
+                    .Callback<ModelMetadata, IEnumerable<Attribute>>((m, attributes) => callbackAttributes = attributes)
+                    .Returns((IEnumerable<IModelValidator>)null)
+                    .Verifiable();
+
+            // Act
+            provider.Object.GetValidators(metadata);
+
+            // Assert
+            provider.Verify();
+            Assert.True(callbackAttributes.Any(a => a is RangeAttribute));
+            Assert.True(callbackAttributes.Any(a => a is RequiredAttribute));
+        }
+
+        [Fact]
+        public void GetModelMetadataTypeAttributeValidators_SameAttributesOnProperty()
+        {
+            // Arrange
+            IEnumerable<Attribute> callbackAttributes = null;
+            var metadata = _metadataProvider.GetMetadataForProperty(null, typeof(PropertyViewModel), "MetadataAttributes");
+            Mock<TestableAssociatedValidatorProvider> provider = new Mock<TestableAssociatedValidatorProvider> { CallBase = true };
+            provider.Setup(p => p.AbstractGetValidators(metadata, It.IsAny<IEnumerable<Attribute>>()))
+                    .Callback<ModelMetadata, IEnumerable<Attribute>>((m, attributes) => callbackAttributes = attributes)
+                    .Returns((IEnumerable<IModelValidator>)null)
+                    .Verifiable();
+
+            // Act
+            provider.Object.GetValidators(metadata);
+
+            // Assert
+            provider.Verify();
+            Assert.True(callbackAttributes.Any(a => a is RangeAttribute));
+
+            var rangeAttr = (RangeAttribute)callbackAttributes.Where(a => a is RangeAttribute).FirstOrDefault();
```

Comment:
see earlier comments about similar line of code in `AssociatedMetadataProviderTest`